console/v1/types_console_link.go

```diff
@@ -48,6 +48,13 @@ type ApplicationMenuSpec struct {
 type NamespaceDashboardSpec struct {
 	// namespaces is an array of namespace names in which the dashboard link should appear.
 	Namespaces []string `json:"namespaces"`
+	// namespaceSelector is used to match labels on the namespace.
+	// If the labels match, link will be shown for the namespace.
```

Comment:
is this used as an alternative to the explicit list of namespaces above?  or is it an additional filter applied against the list of namespaces?

"if unset, the default is not filtering" makes me think it is the latter, but if so the description (and possibly even the field name) should be clearer.

Reply:
Just one namespace? Not a list?

Reply:
My impression, was that it's a additional filter to the namespace array, @spadgett right ?

> "if unset, the default is not filtering" makes me think it is the latter, but if so the description (and possibly even the field name) should be clearer.

@bparees how about:
```
// matchLabels is used as an additional filter applied against list of namespaces.
// If the labels match, link will be shown for the namespaces.
//
// If unset, dashboard link should appear in all the namespaces from the namespaces array 
//
// +optional
matchLabels *metav1.LabelSelector `json:"matchLabels,omitempty"`
```

Reply:
I'd think it would be an "or" where a project could match either filter. In practice, you'll probably use one or the other.

I agree we should be clear on the behavior in the docs.

Reply:
> I'd think it would be an "or" where a project could match either filter. In practice, you'll probably use one or the other.

hopefully we have a use case for adding this selector, so which implementation makes sense for the use case?

I would agree with @spadgett that i'd probably want the ability to either specify specific namespaces, or the ability to specify a selector to pick my namespaces (or both, unioned together), i doubt i'd specify 10 specific namespaces, then setup a selector to filter out some of them (why wouldn't i have just not specified them in the first place?)

The next question is, is a single selector sufficient or should it be an array of selectors?  (what i think @benjaminapetersen may have been asking above)

Reply:
+1 yes, that is my question.

Reply:
> hopefully we have a use case for adding this selector, so which implementation makes sense for the use case?

Yes, this is for a link to Kiali. They would only use the label selector.

> The next question is, is a single selector sufficient or should it be an array of selectors? (what i think @benjaminapetersen may have been asking above)

I was thinking this would just be a map of name value pairs.

Reply:
i think the newer selector api is more sophisticated than just key/value pairs:
https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements

a map of key/values doesn't allow for expression like "namespaces that have both label a=b and c=d, and also namespaces with label e=f"

Reply:
Although having said that, we have done some of the work to support project overview. Maybe it's not too bad.

https://github.com/openshift/console/blob/master/frontend/public/module/k8s/label-selector.js

Reply:
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#labelselector-v1-meta

Reply:
ok, well even if you just do key/value pair matching, i still think you need an array of maps.  Otherwise you're stuck either enforcing "all namespaces must have all of these labels" or "all namespaces must have at least one of these labels" with no room for "namespaces that have 2 of these 3 labels"

Reply:
@spadgett @jhadvig doesn't seem like this dicussion reached resolution?